### ISSUE TITLE
CoreXY Speed at 45degrees?

### ISSUE THREAD

**Issue by changedsoul** [KevinOConnor/klipper#3416]
I have a coreXY configuration with TMC5160 drivers. I noticed my motors switching in and out of StealthChop while it was doing a infill pattern at a 45deg. The programmed speed was 80mm/s, and I have "stealthchop_threshold: 100" for both X and Y drivers.

In a CoreXY config, only one motor moves when going at a 45 deg so I am confused at why it would be switching in and out of stealthChop if my threshold was at 100.
If it were moving only in the Y or X Axis, I would be expecting both motors to be moving slower than programed speed since how the coreXY works. So in a 45 deg I dont understand why its switching out before it reaches its threshold.
I have a strange suspicion the value in the config gets modified to account for coreXY setups? 
Maybe the stealthchop_threshold: 100 gets divided by SQRT(2) when the config is loaded? 
I dont know, just grasping at straws.

Can someone help shed some light on this?

**Comment by klipper-gitissuebot** [comment]
Hi @changedsoul,

It did not look like there was a Klipper log file attached to this ticket. The log file has been engineered to answer common questions the Klipper developers have about the software and its environment (software version, hardware type, configuration, event timing, and hundreds of other questions).

Unfortunately, too many people have opened tickets without providing the log. That consumes developer time; time that would be better spent enhancing the software. If this ticket references an event that has occurred while running the software then the Klipper log must be attached to this ticket. Otherwise, this ticket will be automatically closed in a few days.

For information on obtaining the Klipper log file see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

The log can still be attached to this ticket - just add a comment and attach the log to that comment.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by changedsoul** [comment]
klipper.log
[klippy.log](https://github.com/KevinOConnor/klipper/files/5360211/klippy.log)

**Comment by KevinOConnor** [comment]
> The programmed speed was 80mm/s, and I have "stealthchop_threshold: 100" for both X and Y drivers.

Toolhead speeds are relative to the print.  The stealthchop_threshold is relative to the speed of the stepper motor itself.  If you requested a diagonal move at 80mm/s on corexy, the stepper would move at 80*sqrt(2) ~= 113.137mm/s to accomplish the move at the requested speed.

Separately, I recommend setting stealthchop_threshold to either zero (to always use spreadcycle) or 999999 (to always use stealthchop).

-Kevin

**Comment by changedsoul** [comment]
Thanks for the response. I am still a bit confused. If having one motor run moving the machine at a 45 causes the motor to run Speed*SQRT(2), then if both are running casing it to move parallel to X or Y, is this situation making the motors move ya programmed speed?

Also, I have heard the same of keeping either in stealth spread, but I have found I get the best of both worlds if my threshold is around 100ish. 
Higher and the switch over causes layer shifts, set to high and I cannot achieve fast rapids. I have been printing for a while now with threshold at 100 and even though I hear the clank switch over, I have had. I layer shifts. I don’t print faster than 80 usually, so if I set my threshold at about 120, that should cover my print being quiet and switch over to spreadcycle for the rapids.
